### PR TITLE
[FLINK-7695] [flip6] Add JobConfigHandler for new RestServerEndpoint

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Add `JobConfigHandler` for the new `RestServerEndpoint`.

## Verifying this change

This change added tests and can be verified as follows:

- JSON marshalling is tested via `JobConfigInfoTest`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobConfigHeaders.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+package org.apache.flink.runtime.rest.messages;
+
+import org.apache.flink.runtime.rest.HttpMethodWrapper;
+
+import org.apache.flink.shaded.netty4.io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ *
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobConfigHeaders.java]: empty javadoc

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraphTest.java [hunk 2]
```diff
@@ -142,6 +143,8 @@ public static void setupExecutionGraph() throws Exception {
 
 		Execution executionWithAccumulators = runtimeGraph.getJobVertex(v1ID).getTaskVertices()[0].getCurrentExecutionAttempt();
 
+		runtimeGraph.setJsonPlan("{}");
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraphTest.java]: this seems unrelated. Probably part of porting the JobPlanHandler?

Comment (tillrohrmann) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraphTest.java]: This PR introduces null checks in the `ArchivedExecutionGraph` and that's why this change is included. I could also move these changes to the #4727 PR.

File: flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobConfigHeaders.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/JobConfigHeaders.java]: Good catch. Will add it.

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraphTest.java (hunk 2 above)
Comment (zentol) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraphTest.java]: ah ok, nevermind then.

**Comment by zentol** [comment]
+1 after the javadoc was added.

**Comment by yew1eb** [comment]
LGTM.

**Comment by yew1eb** [comment]
Hey @tillrohrmann, Will you merge this PR ?
I want to fix sibling issues (FLINK-7704, FLINK-7705, FLINK-7706).
:beers:

Best,
Hai Zhou

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol and @yew1eb. I have to throw this PR once more on Travis to see if everything passes. Once this is done and the depended PRs are merged, I'll merge this one as well.
